pkg/util/names/names_test.go

```diff
@@ -100,3 +114,193 @@ func TestGenerateBindingReferenceKey(t *testing.T) {
+
+	for _, test := range tests {
+		got := GenerateWorkName(test.kind, test.name, test.namespace)
+
+		hash := fnv.New32a()
+		hashutil.DeepHashObject(hash, test.workname)
+		if result := fmt.Sprintf("%s-%s", test.name, rand.SafeEncodeString(fmt.Sprint(hash.Sum32()))); result != got {
+			t.Errorf("Test %s failed: expected %v, but got %v", test.testCase, result, got)
+		}
+	}
+}
+
+func TestGenerateServiceAccountName(t *testing.T) {
+	tests := []struct {
+		name        string
+		clusterName string
+		expected    string
+	}{
+		{
+			name:        "non empty clusterName",
+			clusterName: "cluster1",
+			expected:    "karmada-cluster1",
+		},
+	}
+	for _, test := range tests {
+		got := GenerateServiceAccountName(test.clusterName)
+		if got != test.expected {
+			t.Errorf("Test %s failed: expected %v, but got %v", test.name, test.expected, got)
+		}
+	}
+}
+
+func TestGenerateRoleName(t *testing.T) {
+	tests := []struct {
+		name               string
+		serviceAccountName string
+		expected           string
+	}{
+		{
+			name:               "non empty serviceAccountName",
+			serviceAccountName: "account",
+			expected:           "karmada-controller-manager:account",
+		},
+	}
+	for _, test := range tests {
+		got := GenerateRoleName(test.serviceAccountName)
+		if got != test.expected {
+			t.Errorf("Test %s failed: expected %v, but got %v", test.name, test.expected, got)
+		}
+	}
+}
+
+func TestGenerateEndpointSliceName(t *testing.T) {
+	tests := []struct {
+		name              string
+		endpointSliceName string
+		cluster           string
+		expected          string
+	}{
+		{
+			name:              "",
+			endpointSliceName: "endpoint",
+			cluster:           "cluster",
+			expected:          "imported-cluster-endpoint",
+		},
+	}
+	for _, test := range tests {
+		got := GenerateEndpointSliceName(test.endpointSliceName, test.cluster)
+		if got != test.expected {
+			t.Errorf("Test %s failed: expected %v, but got %v", test.name, test.expected, got)
+		}
+	}
+}
+
+func TestGenerateDerivedServiceName(t *testing.T) {
+	tests := []struct {
+		name        string
+		serviceName string
+		expected    string
+	}{
+		{
+			name:        "",
+			serviceName: "service",
+			expected:    "derived-service",
+		},
+	}
+	for _, test := range tests {
+		got := GenerateDerivedServiceName(test.serviceName)
+		if got != test.expected {
+			t.Errorf("Test %s failed: expected %v, but got %v", test.name, test.expected, got)
+		}
+	}
+}
+
+func TestGenerateEstimatorServiceName(t *testing.T) {
+	tests := []struct {
+		name        string
+		clusterName string
+		expected    string
+	}{
+		{
+			name:        "",
+			clusterName: "cluster",
+			expected:    "karmada-scheduler-estimator-cluster",
+		},
+	}
+	for _, test := range tests {
+		got := GenerateEstimatorServiceName(test.clusterName)
+		if got != test.expected {
+			t.Errorf("Test %s failed: expected %v, but got %v", test.name, test.expected, got)
+		}
+	}
+}
+
+func TestGenerateClusterAPISecretName(t *testing.T) {
```

Comment:
Here will conflict with #1029 , because that pr delete the func `GenerateClusterAPISecretName`.

Reply:
Then, let's move this forward after #1029.

Reply:
#1029 get merged.

Reply:
> 

fixed.